tests/cpp/operator/mkldnn_operator_test.cc

```diff
@@ -558,31 +607,41 @@ void TestOpEx(const OpAttrs &forward_attrs, const OpAttrs &backwards_attrs) {
         Engine::Get()->WaitForAll();
         AssertEqual(outputs, ex_outputs);
 
-        // backwards test performed same time since output needed
-        backwards_input[0] = outputs[0];  // output grad
-        backwards_input[1] = inputs[0];  // input
-        backwards_input[2] = outputs[1];  // out norm
+        if (!backwards_attrs.requests.empty()) {
+          TestOpExBackward(forward_attrs, backwards_attrs, OpReqType::kWriteTo,
```

Comment:
If you test backward in a separate function, I suggest to do the same for forward as well to make the test function more logical

Reply:
This function TestOpEx is basically test forward which handles multiple request types like WriteTo and WriteInPlace. For some operators backward test needs to be performed for their specific request types. That is when test backwards is called, the inputs to which are the outputs of test forward. Hence, it's being called from within this function.

Reply:
Can we separate in multiple functions then? It seems this function becomes too bulky at the current design.

Reply:
I think this is a good suggestion for refactoring. Basically since we are doing InvokeOp and Comparison of expected and actual outputs twice in this function can we pull it to its own function like in TestOpExBackward. WDYT?

Reply:
After discussion offline, I understood the test pattern here and therefore approve this change.

Reply:
@anirudh2290 TestOpExBackward is written for only one request type whereas TestOpEx is written for multiple request types which have very few lines of code in common. Hence, refactoring won't make much of a difference here.